repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/accesscontrol/AccessControlPreProcessor.java

```diff
@@ -0,0 +1,111 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.atlas.repository.store.graph.v2.preprocessor.accesscontrol;
+
+import org.apache.atlas.accesscontrol.persona.AtlasPersonaService;
+import org.apache.atlas.accesscontrol.persona.PersonaContext;
+import org.apache.atlas.accesscontrol.purpose.AtlasPurposeService;
+import org.apache.atlas.accesscontrol.purpose.PurposeContext;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.model.instance.AtlasEntity;
+import org.apache.atlas.model.instance.AtlasStruct;
+import org.apache.atlas.model.instance.EntityMutations;
+import org.apache.atlas.repository.graphdb.AtlasGraph;
+import org.apache.atlas.repository.graphdb.AtlasVertex;
+import org.apache.atlas.repository.store.graph.v2.EntityGraphRetriever;
+import org.apache.atlas.repository.store.graph.v2.EntityMutationContext;
+import org.apache.atlas.repository.store.graph.v2.preprocessor.PreProcessor;
+import org.apache.atlas.repository.store.graph.v2.preprocessor.glossary.CategoryPreProcessor;
+import org.apache.atlas.type.AtlasTypeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.atlas.repository.Constants.PERSONA_ENTITY_TYPE;
+import static org.apache.atlas.repository.Constants.QUALIFIED_NAME;
+
+public class AccessControlPreProcessor implements PreProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(CategoryPreProcessor.class);
+
+    private final AtlasTypeRegistry typeRegistry;
+    private final EntityGraphRetriever entityRetriever;
+    private final AtlasPersonaService personaService;
+    private final AtlasPurposeService purposeService;
+
+    public AccessControlPreProcessor(AtlasTypeRegistry typeRegistry, AtlasGraph graph, EntityGraphRetriever entityRetriever) {
+        this.entityRetriever = entityRetriever;
+        this.typeRegistry = typeRegistry;
+        personaService = new AtlasPersonaService(graph, entityRetriever);
+        purposeService = new AtlasPurposeService(graph, entityRetriever);
+    }
+
+    @Override
+    public void processAttributes(AtlasStruct entityStruct, EntityMutationContext context, EntityMutations.EntityOperation operation) throws AtlasBaseException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("AccessControlPreProcessor.processAttributes: pre processing {}, {}",
+                    entityStruct.getAttribute(QUALIFIED_NAME), operation);
+        }
+
+        AtlasEntity entity = (AtlasEntity) entityStruct;
+        AtlasVertex vertex = context.getVertex(entity.getGuid());
+
+        switch (operation) {
```

Comment:
what about other operations like PARTIAL_UPDATE, DELETE, PURGE? Should we add a default case?

Reply:
agree

Reply:
replied [here](https://github.com/atlanhq/atlas-metastore/pull/850/files/1bb8c331e9a252dcb5a7918776c442536f6ad869#r973734762)